docs/src/main/asciidoc/security.adoc

```diff
@@ -3,161 +3,185 @@ This guide is maintained in the main Quarkus repository
+This filter can be registered with MP RestClient implementations injected into the current Quarkus endpoint, which in turn must be protected by using the Quarkus OpenID Connect adapter. 
+This filter can be used to propagate the access token to the downstream services.
 
-`quarkus-smallrye-jwt` provides Microprofile JWT 1.1.1 implementation and many more options to verify signed and encrypted `JWT` tokens and represent them as `org.eclipse.microprofile.jwt.JsonWebToken`.
+For more information, see the xref:security-openid-connect-client.adoc[OpenID Connect client and token propagation quickstart] and xref:security-openid-connect-client-reference.adoc[OpenID Connect (OIDC) and OAuth2 client and filters reference] guides.
 
-It provides an alternative to `quarkus-oidc` Bearer Token Authentication Mechanism. It can currently verify only `JWT` tokens using the PEM keys or refreshable `JWK` key set.
+[[smallrye-jwt]]
+=== SmallRye JWT authentication
 
-Additionally, it provides `JWT Generation API` for creating `signed`, `inner-signed` and/or `encrypted` `JWT` tokens with ease.
+The `quarkus-smallrye-jwt` extension provides a MicroProfile JSON Web Token (JWT) 1.1.1 implementation and multiple options to verify signed and encrypted `JWT` tokens and represents them as `org.eclipse.microprofile.jwt.JsonWebToken`.
 
-See the xref:security-jwt.adoc[Using SmallRye JWT] guide for more information.
+`quarkus-smallrye-jwt` is an alternative to the `quarkus-oidc` Bearer Token authentication mechanism, and verifies only `JWT` tokens by using either PEM keys or the refreshable `JWK` key set.
+`quarkus-smallrye-jwt` also provides the JWT generation API, which you can use to easily create `signed`, `inner-signed`, and `encrypted` `JWT` tokens.
 
-=== OAuth2
+For more information, see xref:security-jwt.adoc[Using SmallRye JWT role-based access control].
 
-`quarkus-elytron-security-oauth2` provides an alternative to `quarkus-oidc` Bearer Token Authentication Mechanism. It is based on `Elytron` and is primarily meant for introspecting the opaque tokens remotely.
+=== OAuth2 authentication
 
-See the xref:security-oauth2.adoc[Using OAuth2] guide for more information.
+`quarkus-elytron-security-oauth2` provides an alternative to the `quarkus-oidc` Bearer Token authentication mechanism. `quarkus-elytron-security-oauth2` is based on `Elytron` and is primarily intended for introspecting opaque tokens remotely.
+For more information, see xref:security-oauth2.adoc[Using OAuth2].
 
 [[oidc-jwt-oauth2-comparison]]
-=== Choosing between OpenID Connect, SmallRye JWT and OAuth2 extensions
+=== Choosing between OpenID Connect, SmallRye JWT, and OAuth2 extensions
+
+Use the following information to help you to decide which authentication extension to use to secure your Quarkus applications:
```

Comment:
@sberyozkin I switched the separate paragraphs to an unordered list. Does this work? 
I wasn't 100% sure if each list item is parallel and works. Please take a look. Thanks :-)

Reply:
@michelle-purcell Thanks for going for carefully going through this section :-), LGTM